core/json/src/main/scala/net/liftweb/json/JsonParser.scala

```diff
@@ -83,89 +85,118 @@ object JsonParser {
       case e: Exception => throw new ParseException("parsing failed", e)
     } finally { buf.release }
   }
-  
-  private[json] def unquote(string: String): String = 
+
+  private[this] final val HexChars: Array[Int] = {
+    val chars = new Array[Int](128)
+    var i = 0
+    while (i < 10) {
+      chars(i + '0') = i
+      i += 1
+    }
+    i = 0
+    while (i < 16) {
+      chars(i + 'a') = 10 + i
+      chars(i + 'A') = 10 + i
+      i += 1
+    }
+    chars
+  }
+
+  private[json] def unquote(string: String): String =
     unquote(new JsonParser.Buffer(new java.io.StringReader(string), false))
-  
-  private[json] def unquote(buf: JsonParser.Buffer): String = {
-    def unquote0(buf: JsonParser.Buffer, base: String): String = {
-      val s = new java.lang.StringBuilder(base)
+
+  private[this] def unquote(buf: JsonParser.Buffer): String = {
+    def unquote0(buf: JsonParser.Buffer): String = {
+      val builder = buf.builder
+      builder.delete(0, builder.length())
       var c = '\\'
       while (c != '"') {
         if (c == '\\') {
-          buf.next match {
-            case '"'  => s.append('"')
-            case '\\' => s.append('\\')
-            case '/'  => s.append('/')
-            case 'b'  => s.append('\b')
-            case 'f'  => s.append('\f')
-            case 'n'  => s.append('\n')
-            case 'r'  => s.append('\r')
-            case 't'  => s.append('\t')
+          buf.substring(intoBuilder = true)
+          (buf.next: @switch) match {
```

Comment:
No, I believe it's [like `tailrec`](http://www.scala-lang.org/api/current/scala/annotation/switch.html).